common/views/components/ImageViewer/ImageViewerImage.js

```diff
@@ -59,13 +64,20 @@ type Props = {|
 
 const ImageViewerImage = ({ id, infoUrl }: Props) => {
   const [scriptError, setScriptError] = useState(false);
+  const [viewer, setViewer] = useState(null);
   const handleScriptError = () => {
     setScriptError(true);
   };
 
   useEffect(() => {
-    setupViewer(infoUrl, id, handleScriptError);
-  }, []);
+    if (viewer) {
+      viewer.destroy();
+    }
+
+    setupViewer(infoUrl, id, handleScriptError).then(viewer => {
```

Comment:
You could use await or `setupViewer(infoUrl, id, handleScriptError).then(setViewer)`.
Neither's better, just shorter.

Reply:
`useEffect(async () => …` isn't supported and feels like overkill to write an extra async function to wrap `setupViewer`. But I like the shorter `.then(setViewer)`

Reply:
Ah yes, sorry my bad.